--- a/Cargo.toml
+++ b/Cargo.toml
@@ -775,3 +775,11 @@ large_types_passed_by_value = "allow"
 
 [workspace.metadata.cargo-shear]
 ignored = ["clap", "fluent", "libstdbuf"]
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../../chrono-0.4.43" }
+iana-time-zone = { path = "../../iana-time-zone-0.1.64" }
+# all: no SYS_getrandom
+getrandom2 = { path = "../../getrandom-0.2.17", package = "getrandom", version = "0.2.17" }
+getrandom4 = { path = "../../getrandom-0.4.2", package = "getrandom", version = "0.4.2" }
