--- a/vendor/sysinfo-0.38.2/src/unix/linux/system.rs
+++ b/vendor/sysinfo-0.38.2/src/unix/linux/system.rs
@@ -410,6 +410,7 @@ impl SystemInner {
     pub(crate) fn name() -> Option<String> {
         get_system_info_linux(
             InfoType::Name,
+            Path::new("/opt/etc/os-release"), // Entware
             Path::new("/etc/os-release"),
             Path::new("/etc/lsb-release"),
         )
@@ -505,6 +506,7 @@ impl SystemInner {
     pub(crate) fn os_version() -> Option<String> {
         get_system_info_linux(
             InfoType::OsVersion,
+            Path::new("/opt/etc/os-release"), // Entware
             Path::new("/etc/os-release"),
             Path::new("/etc/lsb-release"),
         )
@@ -519,6 +521,7 @@ impl SystemInner {
     pub(crate) fn distribution_id() -> String {
         get_system_info_linux(
             InfoType::DistributionID,
+            Path::new("/opt/etc/os-release"), // Entware
             Path::new("/etc/os-release"),
             Path::new(""),
         )
@@ -538,6 +541,7 @@ impl SystemInner {
     pub(crate) fn distribution_id_like() -> Vec<String> {
         system_info_as_list(get_system_info_linux(
             InfoType::DistributionIDLike,
+            Path::new("/opt/etc/os-release"), // Entware
             Path::new("/etc/os-release"),
             Path::new(""),
         ))
