--- a/Cargo.toml
+++ b/Cargo.toml
@@ -66,3 +66,9 @@ serde_test = { version = "1.0.176" }
 # our own crates used as dependencies, same version as the workspace version
 # NOTE: keep this part at the bottom of the file, do not change this line
 ntp-proto = { version = "1.9.0", path = "./ntp-proto", default-features = false, features = ["__internal-api"] }
+
+[patch.crates-io]
+# all: no SYS_getrandom
+getrandom2 = { path = "../getrandom-0.2.17", package = "getrandom", version = "0.2.17" }
+getrandom3 = { path = "../getrandom-0.3.3", package = "getrandom", version = "0.3.3" }
+getrandom4 = { path = "../getrandom-0.4.2", package = "getrandom", version = "0.4.2" }
