--- a/Cargo.toml
+++ b/Cargo.toml
@@ -43,7 +43,7 @@ base64 = "0.22"
 bitflags = "2"
 bytesize = "2"
 cbc = { version = "0.2", features = ["alloc"] }
-chrono = "0.4"
+chrono = "=0.4.44"
 content_inspector = "0.2"
 dirs = "6"
 edit = "0.1"
@@ -54,7 +54,8 @@ log = "0.4"
 md-5 = "0.11"
 notify = "8"
 notify-rust = { version = "4", default-features = false, features = ["d"] }
-nucleo = "0.5"
+# error[E0432]: unresolved import `std::sync::atomic::AtomicU64`
+nucleo = "=0.5.0"
 open = "5"
 rand = "0.10"
 regex = "1"
@@ -62,8 +63,9 @@ remotefs = "0.3"
 remotefs-aws-s3 = "0.4"
 remotefs-kube = "0.4"
 remotefs-smb = { version = "0.3", optional = true }
-remotefs-ssh = { version = "0.8", default-features = false, features = [
-  "russh",
+# keep
+remotefs-ssh = { version = "=0.6.0", default-features = false, features = [
+  "ssh2-vendored",
 ] }
 remotefs-webdav = "0.2"
 rpassword = "7"
@@ -78,7 +80,8 @@ semver = "1"
 serde = { version = "1", features = ["derive"] }
 shellexpand = "3"
 simplelog = "0.12"
-ssh2-config = "0.7"
+# keep
+ssh2-config = "=0.4.0"
 tempfile = "3"
 thiserror = "2"
 tokio = { version = "1", features = ["rt"] }
@@ -145,3 +148,14 @@ incremental = true
 
 [profile.release]
 strip = true
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.44" }
+iana-time-zone = { path = "../iana-time-zone-0.1.64" }
+# mips/mipsel: error[E0432]: unresolved import `std::sync::atomic::AtomicU64`
+nucleo = { path = "../nucleo-0.5.0" }
+# all: no SYS_getrandom
+getrandom2 = { path = "../getrandom-0.2.17", package = "getrandom", version = "0.2.17" }
+getrandom3 = { path = "../getrandom-0.3.4", package = "getrandom", version = "0.3.4" }
+getrandom4 = { path = "../getrandom-0.4.2", package = "getrandom", version = "0.4.2" }
